docs-src/concepts/what-is-a-workflow-definition.md

```diff
@@ -8,15 +8,136 @@ tags:
   - explanation
 ---
 
-A Workflow Definition is the code that defines the constraints of a Workflow Execution.
+A Workflow Definition is the code that defines the Workflow and written with a programming language and a corresponding Temporal SDK.
+A Workflow Definition defines the constraints of a Workflow Execution.
+
+Below are different ways to develop a basic Workflow Definition.
+
+<Tabs groupId="basic-workflow-definition" queryString>
+<TabItem value="go" label="Go">
+
+**Workflow Definition in Go**
+
+```go
+package yourapp
+
+import (
+	"time"
+
+	"go.temporal.io/sdk/workflow"
+)
+// ...
+
+// YourSimpleWorkflowDefintiion is the most basic Workflow Defintion.
+func YourSimpleWorkflowDefinition(ctx workflow.Context) error {
+	// ...
+	return nil
+}
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+**Workflow Definition in Java**
+
+```java
+// Workflow interface
+@WorkflowInterface
+public interface YourWorkflow {
+
+    @WorkflowMethod
+    String yourWFMethod(Arguments args);
+}
+```
+
+</TabItem>
+<TabItem value="php" label="PHP">
+
+**Workflow Definition in PHP**
+
+```PHP
+use Temporal\Workflow\YourWorkflowInterface;
+use Temporal\Workflow\WorkflowMethod;
+
+#[WorkflowInterface]
+interface FileProcessingWorkflow
+{
+    #[WorkflowMethod]
+    public function processFile(Argument $args);
+
+}
+```
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+**Workflow Definition in Python**
+
+```Python
+from temporalio import workflow
+# ...
+# ...
+@workflow.defn(name="YourWorkflow")
+class YourWorkflow:
+    @workflow.run
+    async def run(self, name: str) -> str:
+        return await workflow.execute_activity(
+            your_activity,
+            YourParams("Hello", name),
+            start_to_close_timeout=timedelta(seconds=10),
+        )
```

Comment:
The go example glosses over API calls and Activities and just returns - perhaps we want the same here?